File: backend/src/main/java/com/apocalypse/thefall/mapper/character/stats/DamageTypeMapper.java
package com.apocalypse.thefall.mapper.character.stats;

import com.apocalypse.thefall.dto.character.stats.DataItemDto;
import com.apocalypse.thefall.entity.item.DamageType;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface DamageTypeMapper {

    DataItemDto toDto(DamageType damageType);

    List<DataItemDto> toDto(List<DamageType> damageTypes);
}

File: backend/src/main/java/com/apocalypse/thefall/mapper/character/stats/damagetype/DamageTypeMapper.kt
package com.apocalypse.thefall.mapper.character.stats.damagetype

import com.apocalypse.thefall.dto.character.stats.DataItemDto
import com.apocalypse.thefall.entity.item.DamageType

fun DamageType.toDataItemDto(): DataItemDto =
    DataItemDto(
        id = id,
        code = code.name,
        names = names,
        descriptions = descriptions,
        imagePath = imagePath,
        displayOrder = displayOrder,
        visible = visible,
        shortNames = null,
        camelCaseCode = code.name
    )

fun List<DamageType>.toListDataItemDto(): List<DataItemDto> =
    map { it.toDataItemDto() }
